utils/build-script

```diff
@@ -429,11 +429,12 @@ class BuildScriptInvocation(object):
                 "can't find source directory for ninja "
                 "(tried %s)" % (self.workspace.source_dir("ninja")))
 
-        ninja_build = products.Ninja(
+        ninja_build = products.Ninja.make_builder(
             args=self.args,
             toolchain=self.toolchain,
-            source_dir=self.workspace.source_dir("ninja"),
-            build_dir=self.workspace.build_dir("build", "ninja"))
+            workspace=self.workspace,
+            host=StdlibDeploymentTarget.get_target_for_name(
```

Comment:
None of this is your doing, I'm wondering if you know the history.  Why is this in `StdlibDeploymentTarget`?  Why is this being built for the `host_target` and not `build_target`?  This is the tool being built for the build right?

Reply:
I think `StdlibDeploymentTarget` is a little misnamed at this point, because it is used for more things that just the stdlib. If you think about it as `CompilationTarget` or `DeploymentTarget` is easier to understand.

Ninja is build only for the machine building the rest of the pieces (it is build if it cannot be found in the toolchain/path), so it is fine to “hardcode” here to `host_target` (which is the current machine).

If you look at the implementation of `make_builder`, it actually ignores the target. The previous implementation didn’t care about the target (so it builds for the local target).

PD: there's a big mess of names in the build script. Sometimes `host` refers to the device that run the code (arguments `--host-test` and `--skip-test-ios-host` which control to run or not tests on the external devices), but sometimes refers to the machine doing the build (like this `host_target` and arguments like `--build-runtime-with-host-compiler` and `--host-cc`).

Reply:
Right, I figured it was a poorly named variable at this point.  I was thinking of it as `TargetInformation`.  I was wondering if you knew why this had come to be poorly named at this point.

As to the Ninja bits, right - its again a name thing.  `target` is the target where generated code will be run, `host` is where the generated code generator will run, and `build` is the machine where the build is occurring.  These are well documented names, so it was a question of does the current machinery not name things accordingly?

I think that making it deal with cross-compilation is great thing, but way outside the scope of this current change .

Reply:
About the renaming, from comments and clues in the code, I think there was a previous try to make everything more standard, but it was necessary to keep the argument names the same for compatibility, which probably makes things a little bit more complicated. I can see if something can be done as a cleanup when all these pieces fall into place (I prefer to have the “wrong” names, since they match better what the `build-script-impl` have).